### PR TITLE
planner: fix the wrong cost formula of MPPExchanger

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #35240

Problem Summary: fix the wrong cost formula of MPPExchanger

### What is changed and how it works?

fix the wrong cost formula of MPPExchanger, the net cost part should be `rows*row-size*net-factor` instead of `rows*net-factor`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by time-and-fate — APPROVED** [review]

File: executor/tiflashtest/tiflash_test.go [hunk 1]
```diff
@@ -281,7 +281,7 @@ func TestMppExecution(t *testing.T) {
 	require.NoError(t, err)
 	ts := txn.StartTS()
 	taskID := tk.Session().GetSessionVars().AllocMPPTaskID(ts)
-	require.Equal(t, int64(6), taskID)
```
Comment (qw4990) [inline executor/tiflashtest/tiflash_test.go:284]: The plan changed.

**Review by Yisaer — APPROVED** [review]

**Pull request merged**